projects/addon-mobile/directives/elastic-sticky/elastic-sticky.service.ts

```diff
@@ -1,41 +1,41 @@
-import {ElementRef, inject, Injectable, NgZone} from '@angular/core';
+import {
+    afterNextRender,
+    ChangeDetectorRef,
+    ElementRef,
+    inject,
+    Injectable,
+    INJECTOR,
+    NgZone,
+} from '@angular/core';
 import {
     TuiDestroyService,
     tuiGetElementOffset,
     tuiScrollFrom,
-    tuiZoneOptimized,
+    tuiZonefree,
 } from '@taiga-ui/cdk';
 import {SCROLL_REF_SELECTOR, TUI_SCROLL_REF} from '@taiga-ui/core';
-import {
-    distinctUntilChanged,
-    map,
-    Observable,
-    skip,
-    startWith,
-    switchMap,
-    take,
-    takeUntil,
-} from 'rxjs';
+import {map, Observable, takeUntil, tap} from 'rxjs';
 
 @Injectable()
 export class TuiElasticStickyService extends Observable<number> {
+    private readonly injector = inject(INJECTOR);
     private readonly el: HTMLElement = inject(ElementRef).nativeElement;
     private readonly scrollRef: HTMLElement = inject(TUI_SCROLL_REF).nativeElement;
     private readonly zone = inject(NgZone);
+    private readonly cd = inject(ChangeDetectorRef);
     private readonly destroy$ = inject(TuiDestroyService, {self: true});
 
     constructor() {
-        super(subscriber =>
-            this.zone.onStable
-                .pipe(
-                    take(1),
-                    switchMap(() => {
-                        const host =
-                            this.el.closest(SCROLL_REF_SELECTOR) || this.scrollRef;
-                        const {offsetHeight} = this.el;
-                        const {offsetTop} = tuiGetElementOffset(host, this.el);
+        super(subscriber => {
+            afterNextRender(
+                () => {
+                    const host = this.el.closest(SCROLL_REF_SELECTOR) || this.scrollRef;
+                    const {offsetHeight} = this.el;
+                    const {offsetTop} = tuiGetElementOffset(host, this.el);
 
-                        return tuiScrollFrom(host).pipe(
+                    return tuiScrollFrom(host)
```

Comment:
I don't think this return matters. We are supposed to return teardown logic but there is void returned from the super callback.